### DISCUSSION TITLE
socket channel WIKI 文档有点旧， 能否更新一下

### DISCUSSION THREAD

**Discussion by aceyin** [cloudwu/skynet#1807] [General]
关于 socket channel 官方的文档主要就是这一篇了：`https://github.com/cloudwu/skynet/wiki/SocketChannel`
其中关于 `response` 函数的说明如下：

`
response 函数的第一个返回值需要是一个 boolean ，如果为 true 表示协议解析正常；如果为 false 表示协议出错，这会导致连接断开且让 request 的调用者也获得一个 error 。
`

但翻看源代码之后， 发现源码里面的返回值并非这样的，而是这样的：
https://github.com/cloudwu/skynet/blob/5ec02a4a2c5bb42e2f7c9165f6132e4252e6f8a0/lualib/skynet/socketchannel.lua#L96

也就是说该 response 函数， 实际上需要返回： session_id, ok, data, padding 4 个值

所以，能否更新一下WIKI上面的文档？

**Comment by CoolDesert** [comment]
wiki 任何人都可以编辑 ，发现错误可以自行修改。

**Comment by cloudwu** [comment]
你看错了代码，对应 wiki 上的代码是：
https://github.com/cloudwu/skynet/blob/master/lualib/skynet/socketchannel.lua#L201

你说的 4 个值的用法，wiki 文档后半段有说明，请注意阅读。

btw,  issue 只用于追踪代码的 bug ，其它问题请提到 discussions 。若不确定是提 issue 还是 discussion ，默认请用 discussion 。
